### PR TITLE
Add dedicated macOS resource doc

### PR DESCRIPTION
**Pull request by kbatuigas**
# Description
We have a new dedicated macOS resource class, which provides an XL macOS for our customers, with increased storage and a 24-hour individual job timeout.

The new docs have currently been added to an existing `hello-world-macos` page just as a draft/proof of concept but this can still be moved to a standalone page.

### CODE REVIEW

File: jekyll/_cci2/hello-world-macos.md [hunk 1]
```diff
@@ -156,6 +158,74 @@ straightforward. To build ARM64 binaries, prepend the `xcodebuild` command with
...
 
+
+## Dedicated Hosts for macOS
+{: #dedicated-hosts-macos }
+
+The dedicated host resource class is a new macOS option for those developing, building, testing, and signing iOS, iPadOS, macOS, WatchOS, and tvOS applications using the Xcode IDE. These dedicated resources provide an isolated environment for increased security.
+
+This resource class requires a 24-hour minimum lease and runs on Intel-based Mac hardware.
+
+The identifier for the dedicated host resource is metal and supports the following Xcode images:
```
Comment (devinmarieb) [inline jekyll/_cci2/hello-world-macos.md]: would "metal" need to be `metal`?

File: jekyll/_cci2/hello-world-macos.md [hunk 2]
```diff
@@ -156,6 +158,74 @@ straightforward. To build ARM64 binaries, prepend the `xcodebuild` command with
...
+
+| Resource Class Name  | vCPU  | Memory  | Storage  | Cost  |
+|---|---|---|---|---|
+| `metal`  | 12  | 32 GB  | 200 GB  | 100 credits per minute (24-hour minimum)  |
+{: class="table table-striped"}
+
+### Known limitations
+
+- The resource class does not currently support test splitting or parallelism.
+- The host gets cleaned between jobs, which can take 5-45 minutes currently.
```
Comment (devinmarieb) [inline jekyll/_cci2/hello-world-macos.md]: I would rephrase this so the sentence doesn't end on "currently." Maybe: "The host gets cleaned between jobs, which can currently take between 5-45 minutes."

File: jekyll/_cci2/dedicated-hosts-macos.adoc [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+| 12 
+| 32 GB
+| 200 GB
+| 00 credits per minute (24-hour minimum)
+|===
+
+=== Known limitations
+
+- The resource class does not currently support test splitting or parallelism.
+- The host gets cleaned between jobs, which can take 5-45 minutes currently.
```
Comment (devinmarieb) [inline jekyll/_cci2/dedicated-hosts-macos.adoc:52]: I would rephrase this so the sentence doesn't end on "currently." Maybe: "The host gets cleaned between jobs, which can currently take between 5-45 minutes."

**Review by devinmarieb — APPROVED** [review]
Just one comment about wording, but looks goo!

File: jekyll/_cci2/dedicated-hosts-macos.adoc [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+:toc: macro
+:toc-title:
+
+== Dedicated Hosts for macOS
+
+The dedicated host resource class is a new macOS option for those developing, building, testing, and signing iOS, iPadOS, macOS, WatchOS, and tvOS applications using the Xcode IDE. These dedicated resources provide an isolated environment for increased security.
+
+This resource class requires a 24-hour minimum lease and runs on Intel-based Mac hardware.
+
+The identifier for the dedicated host resource is `metal` and supports the following Xcode images:
```
Comment (alexazl) [inline jekyll/_cci2/dedicated-hosts-macos.adoc:19]: For GA, the identifier will be `macos.x86.metal.gen1`.

File: jekyll/_cci2/dedicated-hosts-macos.adoc [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+- Xcode 12.5.1
+- Xcode 12.4.0
+- Xcode 12.3.0
+- Xcode 12.2.0
+
+== Pricing and specs
+
+Once a dedicated host has been allocated, you have exclusive access to it for a minimum of 24 hours. If the dedicated host is already in use when a job is kicked off, an additional dedicated host is reserved (with its own 24 hour lease window).
+
+Each account can currently have a maximum of three concurrent dedicated hosts. Any time over the initial 24 hours that a dedicated host is in use will be charged a per-minute overage rate (see table below for pricing details).
```
Comment (alexazl) [inline jekyll/_cci2/dedicated-hosts-macos.adoc:31]: The default maximum is now 1. We should add a note that they can increase that amount by reaching out to support or their account rep.

File: jekyll/_cci2/dedicated-hosts-macos.adoc [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+
+- Xcode 13.0.0
+- Xcode 12.5.1
+- Xcode 12.4.0
+- Xcode 12.3.0
+- Xcode 12.2.0
+
+== Pricing and specs
+
+Once a dedicated host has been allocated, you have exclusive access to it for a minimum of 24 hours. If the dedicated host is already in use when a job is kicked off, an additional dedicated host is reserved (with its own 24 hour lease window).
```
Comment (alexazl) [inline jekyll/_cci2/dedicated-hosts-macos.adoc:29]: Since the default is 1, we should update this to say "If the dedicated host is already in use when a job is kicked off, it will queue until the host is available, unless you have increased your dedicated host limit to more than the default of 1." Or something like that.

File: jekyll/_cci2/dedicated-hosts-macos.adoc [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+[.table.table-striped]
+[cols=5*, options="header", stripes=even]
+|===
+| Resource Class Name
+| vCPU
+| Memory
+| Storage
+| Cost
+
+| `metal`
```
Comment (alexazl) [inline jekyll/_cci2/dedicated-hosts-macos.adoc:42]: Update to `macos.x86.metal.gen1`

File: jekyll/_cci2/dedicated-hosts-macos.adoc (hunk 5 above)
Comment (alexazl) [reply jekyll/_cci2/dedicated-hosts-macos.adoc:31]: We also don't need to worry about the overage rate, since we've simplified the pricing to be 100 credits/minute (with a 24-hour minimum).

File: jekyll/_cci2/dedicated-hosts-macos.adoc [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+== Example configuration file using macOS dedicated host resources
+
+```yaml
+# .circleci/config.yml
+version: 2.1
+jobs: 
+  build-and-test: 
+    macos:
+      xcode: 12.5.1 # indicate our selected version of Xcode
+    resource_class: metal # dedicated host, with 24-hour billing
```
Comment (alexazl) [inline jekyll/_cci2/dedicated-hosts-macos.adoc:63]: update resource class tag to `macos.x86.metal.gen1`

File: jekyll/_cci2/dedicated-hosts-macos.adoc [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+          
+workflows:
+  build-test:
+    jobs:
+      - build-and-test
+```
+
+== FAQ
+
+*Q:* Is there an Apple Silicon option for dedicated hosts?
```
Comment (alexazl) [inline jekyll/_cci2/dedicated-hosts-macos.adoc:83]: I actually think we should leave out the Apple Silicon Q/A. We're not quite sure what support will look like at this point.

File: jekyll/_cci2/dedicated-hosts-macos.adoc [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+workflows:
+  build-test:
+    jobs:
+      - build-and-test
+```
+
+== FAQ
+
+*Q:* Why is there a 24-hour minimum?
+
```
Comment (alexazl) [inline jekyll/_cci2/dedicated-hosts-macos.adoc:78]: Should there be a space here? There isn't one for the subsequent Q/A.

File: jekyll/_cci2/dedicated-hosts-macos.adoc [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+:toc: macro
+:toc-title:
+
+== Dedicated Hosts for macOS
+
+The dedicated host resource class is a new macOS option for those developing, building, testing, and signing iOS, iPadOS, macOS, WatchOS, and tvOS applications using the Xcode IDE. These dedicated resources provide an isolated environment for increased security.
+
+This resource class requires a 24-hour minimum lease and runs on Intel-based Mac hardware.
+
+The identifier for the dedicated host resource is `macos.x86.metal.gen1`, and supports the same Xcode images as the other macOS resource classes. Refer to the https://circleci.com/docs/2.0/testing-ios/#supported-xcode-versions[Supported Xcode versions section] of the Testing iOS page for the full list of versions.
```
Comment (alexazl) [inline jekyll/_cci2/dedicated-hosts-macos.adoc:19]: We should add the caveat here that Xcode 10.3 is not supported.

**Review by thomasr8 — APPROVED** [review]

**Review by alexazl — APPROVED** [review]

**Review by devinmarieb — APPROVED** [review]
Yeah I think in this context, an "i" icon is a good choice.
